.github/workflows/build-and-run.yml

```
@@ -0,0 +1,128 @@
+name: Build-And-Run Tests
+
+on: [pull_request, push]
```

>I was thinking that all our branches here will be develop or release branches, so we'll want to run this workflow on all of them

https://github.com/isce-framework/isce3/pull/32#discussion_r2103452198

Hmm, I still think it's correct to constrain which branches this workflow will trigger on.

Even if we ensure that all non-develop, non-release branches are created on forks, this CI workflow will still run on pushes to the forked repo. So, if I push a commit to a PR originating from a fork, the workflow will run twice -- once in the upstream repo and once in my forked repo. This seems like an unnecessary waste of compute resources and also means that I'll get twice as many email notifications if I push a bad commit that causes the CI to fail.

```suggestion
on:
  pull_request:
  push:
    branches:
      - develop
      - release-*
    tags:
      - v*.*.*
```